packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```
@@ -0,0 +1,242 @@
+}): Promise<Result<RelayTradeQuote[] | RelayTradeRate[], SwapErrorRight>> {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+    receiveAddress,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    slippageTolerancePercentageDecimal: _slippageTolerancePercentageDecimal,
+  } = input
+
+  const slippageToleranceBps = _slippageTolerancePercentageDecimal
+    ? convertDecimalPercentageToBasisPoints(_slippageTolerancePercentageDecimal).toFixed()
+    : undefined
+
+  const sellRelayChainId = relayChainMap[sellAsset.chainId]
+  const buyRelayChainId = relayChainMap[buyAsset.chainId]
+
+  if (sellRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (buyRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'sendAddress is required',
+      }),
+    )
+  }
+
+  const maybeQuote = await getQuote(
+    {
+      originChainId: sellRelayChainId,
+      originCurrency: getRelayEvmAssetAddress(sellAsset),
+      destinationChainId: buyRelayChainId,
+      destinationCurrency: getRelayEvmAssetAddress(buyAsset),
+      tradeType: 'EXACT_INPUT',
+      amount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      recipient: receiveAddress,
+      user: sendAddress,
+      slippageTolerance: slippageToleranceBps,
+    },
+    deps.config,
+  )
+
+  if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+
+  const quote = maybeQuote.unwrap()
+
+  const swapSteps = quote.data.steps.filter(step => step.id !== 'approve')
```

blocking: after unwrapping, and before proceeding with parsing logic we'll want to throw a monadic error if `(!quote.details)`.

We'll probably also want to check each of
- `slippageTolerance` (always returned, even if user-provided)
- `rate`
- `currencyOut` 
 
Individually, as those are required quote/rate metadata.

---

Done in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)